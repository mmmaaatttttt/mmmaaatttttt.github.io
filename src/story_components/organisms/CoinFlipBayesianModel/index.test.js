import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { CoinFlipBayesianModel } from ".";

describe("smoke and snapshot tests", () => {
  it("renders successfully", () => {
    shallow(<CoinFlipBayesianModel />);
  });

  it("renders first demonstration successfully", () => {
    const wrapper = shallow(<CoinFlipBayesianModel />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("state changes", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CoinFlipBayesianModel />);
  });

  it("handles switch change", () => {
    wrapper.instance().toggleDistribution();
    expect(wrapper.state().uniform).toBe(false);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("handles heads increase", () => {
    expect(wrapper.state().heads).toBe(0);
    wrapper
      .find("Button")
      .at(0)
      .simulate("click");
    expect(wrapper.state().heads).toBe(1);
  });

  it("handles tails increase", () => {
    expect(wrapper.state().tails).toBe(0);
    wrapper
      .find("Button")
      .at(1)
      .simulate("click");
    expect(wrapper.state().tails).toBe(1);
  });

  it("handles reset", () => {
    wrapper.setState({ heads: 10, tails: 8 });
    wrapper
      .find("Button")
      .at(2)
      .simulate("click");
    expect(wrapper.state()).toMatchObject({ heads: 0, tails: 0 });
  });
});
