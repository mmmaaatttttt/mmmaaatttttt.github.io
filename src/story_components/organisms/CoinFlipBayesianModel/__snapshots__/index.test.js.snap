// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests renders first demonstration successfully 1`] = `
<styled.div
  fullWidthAt="small"
  width="70%"
>
  <ToggleSwitch
    handleSwitchChange={[Function]}
    leftColor="#FF3C23"
    leftText="All probs equally likely"
    rightColor="#5ECFFF"
    rightText="Fair coin more likely"
  />
  <styled.div
    column={false}
    cross="stretch"
    flex={1}
    main="space-evenly"
    margin="0"
    shouldWrap={false}
    textAlign="left"
    width="auto"
  >
    <Button
      color="white"
      disabled={false}
      large={false}
      onClick={[Function]}
    >
      Heads: 
      0
    </Button>
    <Button
      color="white"
      disabled={false}
      large={false}
      onClick={[Function]}
    >
      Tails: 
      0
    </Button>
    <Button
      color="#555555"
      disabled={false}
      large={false}
      onClick={[Function]}
    >
      Reset Counts
    </Button>
  </styled.div>
  <Animate
    show={true}
    start={
      Object {
        "color": "#FF3C23",
        "heads": 0,
        "tails": 0,
      }
    }
    update={
      Object {
        "color": Array [
          "#FF3C23",
        ],
        "heads": Array [
          0,
        ],
        "tails": Array [
          0,
        ],
        "timing": Object {
          "duration": 400,
        },
      }
    }
  >
    <Component />
  </Animate>
</styled.div>
`;

exports[`state changes handles switch change 1`] = `
<styled.div
  fullWidthAt="small"
  width="70%"
>
  <ToggleSwitch
    handleSwitchChange={[Function]}
    leftColor="#FF3C23"
    leftText="All probs equally likely"
    rightColor="#5ECFFF"
    rightText="Fair coin more likely"
  />
  <styled.div
    column={false}
    cross="stretch"
    flex={1}
    main="space-evenly"
    margin="0"
    shouldWrap={false}
    textAlign="left"
    width="auto"
  >
    <Button
      color="white"
      disabled={false}
      large={false}
      onClick={[Function]}
    >
      Heads: 
      0
    </Button>
    <Button
      color="white"
      disabled={false}
      large={false}
      onClick={[Function]}
    >
      Tails: 
      0
    </Button>
    <Button
      color="#555555"
      disabled={false}
      large={false}
      onClick={[Function]}
    >
      Reset Counts
    </Button>
  </styled.div>
  <Animate
    show={true}
    start={
      Object {
        "color": "#FF3C23",
        "heads": 50,
        "tails": 50,
      }
    }
    update={
      Object {
        "color": Array [
          "#5ECFFF",
        ],
        "heads": Array [
          50,
        ],
        "tails": Array [
          50,
        ],
        "timing": Object {
          "duration": 400,
        },
      }
    }
  >
    <Component />
  </Animate>
</styled.div>
`;
